sdk/cosmos/azure-cosmos-encryption/src/test/java/com/azure/cosmos/encryption/CosmosEncryptionItemSerializerTest.java

```diff
@@ -0,0 +1,810 @@
+            } else {
+                return CosmosPatchOperations
+                    .create()
+                    .add("/wrappedContent/someNumber", 555);
+            }
+        };
+
+        runPointOperationAndQueryTestCase(
+            doc,
+            id,
+            onBeforeReplace,
+            onBeforePatch,
+            requestLevelSerializer,
+            TestDocument.class);
+    }
+
+    @Test(groups = { "fast", "emulator" }, dataProvider = "testConfigs_requestLevelSerializer", timeOut = TIMEOUT * 1000000)
+    public void pointOperationsAndQueryWithObjectNode(CosmosItemSerializer requestLevelSerializer) {
+        String id = UUID.randomUUID().toString();
+        ObjectNode doc = TestDocument.createAsObjectNode(id);
+        Consumer<ObjectNode> onBeforeReplace = item -> item.put("someNumber", 999);
+        BiFunction<ObjectNode, Boolean, CosmosPatchOperations> onBeforePatch = (item, isEnvelopeWrapped) -> {
+
+            item.put("someNumber", 555);
+
+            if (!isEnvelopeWrapped) {
+                return CosmosPatchOperations
+                    .create()
+                    .add("/someNumber", 555);
+            } else {
+                return CosmosPatchOperations
+                    .create()
+                    .add("/wrappedContent/someNumber", 555);
+            }
+        };
+
+        runPointOperationAndQueryTestCase(
+            doc,
+            id,
+            onBeforeReplace,
+            onBeforePatch,
+            requestLevelSerializer,
+            ObjectNode.class);
+    }
+
+    private <T> void runPointOperationAndQueryTestCase(
+        T doc,
+        String id,
+        Consumer<T> beforeReplace,
+        BiFunction<T, Boolean, CosmosPatchOperations> beforePatch,
+        CosmosItemSerializer requestLevelSerializer,
+        Class<T> classType) {
+
+        CosmosItemSerializer customSerializer = ImplementationBridgeHelpers
+            .CosmosClientBuilderHelper
+            .getCosmosClientBuilderAccessor()
+            .getDefaultCustomSerializer(this.getClientBuilder());
+
+        boolean useEnvelopeWrapper =
+            requestLevelSerializer == EnvelopWrappingItemSerializer.INSTANCE_NO_TRACKING_ID_VALIDATION
+                || (requestLevelSerializer == null
+                && customSerializer == EnvelopWrappingItemSerializer.INSTANCE_NO_TRACKING_ID_VALIDATION);
+
+        if (requestLevelSerializer == EnvelopWrappingItemSerializer.INSTANCE_NO_TRACKING_ID_VALIDATION
+            && isContentOnWriteEnabled
+            && nonIdempotentWriteRetriesEnabled
+            && useTrackingIdForCreateAndReplace) {
+
+            requestLevelSerializer = EnvelopWrappingItemSerializer.INSTANCE_WITH_TRACKING_ID_VALIDATION;
+        }
+
+        CosmosItemRequestOptions requestOptions = new CosmosItemRequestOptions()
+            .setCustomSerializer(requestLevelSerializer);
+        CosmosItemResponse<T> pojoResponse = container.createItem(doc, new PartitionKey(id), requestOptions);
```

Comment:
maybe also add a test case for createItem without partitionKey as well -> `container.createItem(doc, requestOptions)`

Reply:
Done